wurst/_handles/Dialog.wurst

```diff
@@ -20,4 +23,57 @@ public function dialog.display(player whichPlayer, boolean flag)
     DialogDisplay(whichPlayer, this, flag)
 
 public function dialog.setMessage(string messageText)
-    DialogSetMessage(this, messageText)
\ No newline at end of file
+    DialogSetMessage(this, messageText)
+
+
+public class DialogBox
+
+    private static HashMap<button,trigger> buttonTriggers = new HashMap<button,trigger>
+    private static HashMap<dialog,DialogBox> dialogBoxes = new HashMap<dialog,DialogBox>
+    private dialog dialogHandle
+    private trigger clickDialog
+    private string title
+    private LinkedList<button> buttons
+
+    construct(string title)
+        this.title = title
+        this.dialogHandle = createDialog()
+        this.dialogHandle.setMessage(title)
+        this.clickDialog = CreateTrigger()
+        this.clickDialog.registerDialogEvent(this.dialogHandle)
+        this.clickDialog.addAction(function removeClickedDialog)
+        this.buttons = new LinkedList<button>
+        dialogBoxes.put(this.dialogHandle, this)
+
+    static function removeClickedDialog()
+        button b = GetClickedButton()
+        buttonTriggers.get(b).execute()
+        destroy(dialogBoxes.get(GetClickedDialog()))
+
+    function addButton(string buttonText, integer hotkey, code cb)
+        button b = this.dialogHandle.addButton(buttonText, hotkey)
+        this.buttons.add(b)
+        trigger t = CreateTrigger()
+        t.addAction(cb)
+        buttonTriggers.put(b, t)
+
+    function addButton(string buttonText, code cb)
+        this.addButton(buttonText, 0 , cb)
+
```

Comment:
Add the action that calls ```removeClickedDialog``` on both ```addButton``` functions. This way the user does not have to use it explicitly. Also make ```removeClickedDialog``` a ```private static function```.